ecma402_traits/src/datetimeformat.rs

```diff
@@ -0,0 +1,275 @@
+    pub enum DayPeriod {
+        Narrow,
+        Short,
+        Long,
+    }
+
+    /// Controls the number formatting.
+    ///
+    /// Possible values include: "arab", "arabext", " bali", "beng", "deva", "fullwide", "gujr",
+    /// "guru", "hanidec", "khmr", " knda", "laoo", "latn", "limb", "mlym", " mong", "mymr",
+    /// "orya", "tamldec", " telu", "thai", "tibt".
+    ///
+    /// The value entered as currency is not validated.  This responsibility is
+    /// delegated to the implementor.
+    #[derive(Eq, PartialEq, Debug, Clone)]
+    pub struct NumberingSystem(pub String);
+
+    impl From<&str> for NumberingSystem {
+        fn from(s: &str) -> Self {
+            Self(s.to_string())
+        }
+    }
+
+    impl Default for NumberingSystem {
+        fn default() -> Self {
+            Self("latn".to_string())
+        }
+    }
+
+    /// Controls the time zone formatting.
+    ///
+    /// The value entered as currency is not validated.  This responsibility is
+    /// delegated to the implementor.
+    #[derive(Eq, PartialEq, Debug, Clone)]
+    pub struct TimeZone(pub String);
+
+    impl From<&str> for TimeZone {
+        fn from(s: &str) -> Self {
+            Self(s.to_string())
+        }
+    }
+
+    impl Default for TimeZone {
+        fn default() -> Self {
+            Self("UTC".to_string())
+        }
+    }
+
+    /// The hour cycle to use
+    #[derive(Eq, PartialEq, Debug, Clone)]
+    pub enum HourCycle {
+        /// 12 hour cycle, 0..11.
+        H11,
+        /// 12 hour cycle, 1..12.
+        H12,
+        /// 4 hour cycle, 0..23.
+        H23,
+        /// 4 hour cycle, 1..24.
+        H24
+    }
+
+    #[derive(Eq, PartialEq, Debug, Clone)]
+    pub enum Weekday {
+        /// "Thursday"
+        Long,
+        /// "Thu",
+        Short,
+        /// "T",
+        Narrow,
+    }
+
+    #[derive(Eq, PartialEq, Debug, Clone)]
+    pub enum Era {
+        /// "Anno Domini"
+        Long,
+        /// "AD",
+        Short,
+        /// "A",
+        Narrow,
+    }
+
+    #[derive(Eq, PartialEq, Debug, Clone)]
+    pub enum DisplaySize {
+        /// "2012"
+        Numeric,
+        /// "12"
+        TwoDigit,
+    }
+
+    #[derive(Eq, PartialEq, Debug, Clone)]
+    pub enum Month {
+        /// "3"
+        Numeric,
+        /// "03",
+        TwoDigit,
+        /// "March",
+        Long,
+        /// "Mar"
+        Short,
+        /// "M"
+        Narrow,
+    }
+
+    #[derive(Eq, PartialEq, Debug, Clone)]
+    pub enum TimeZoneStyle {
+    }
+}
+
+#[derive(Eq, PartialEq, Debug, Clone)]
+pub struct DateTimeFormatOptions {
+    /// The formatting style to use for formatting the date part.
+    pub date_style: options::Style,
+    /// The formatting style to use for formatting the time part.
+    pub time_style: options::Style,
+    /// The number of fractional seconds to apply when calling `format`.
+    /// Valid values are 0 to 3.
+    pub fractional_second_digits: u8,
```

Comment:
Should this be strongly typed to enforce valid values?
(Would be nice if Rust had [ranged integers](https://github.com/rust-lang/rfcs/issues/671).)

Reply:
Perhaps, but it's not clear to me how a trait would be doing this.  I'd leave it to the implementor of `new`.